src/validation.py

```diff
@@ -40,6 +40,35 @@ def validate_year(year):
         raise UserWarning(year_warning)
 
 
+def check_allocated_gf_matches_input_gf(pudl_out, gen_fuel_allocated):
+    """Checks that the allocated generation and fuel from EIA-923 matches the input totals."""
+    gf = pudl_out.gf_eia923()
+    plant_total_gf = gf.groupby("plant_id_eia")[
+        [
+            "net_generation_mwh",
+            "fuel_consumed_mmbtu",
+            "fuel_consumed_for_electricity_mmbtu",
+        ]
+    ].sum()
+    plant_total_alloc = gen_fuel_allocated.groupby("plant_id_eia")[
+        [
+            "net_generation_mwh",
+            "fuel_consumed_mmbtu",
+            "fuel_consumed_for_electricity_mmbtu",
+        ]
+    ].sum()
+    # calculate the difference between the values
+    plant_total_diff = plant_total_gf - plant_total_alloc
+    # flag values where the absolute difference is greater than 10 mwh or mmbtu
+    mismatched_allocation = plant_total_diff[
+        (abs(plant_total_diff["fuel_consumed_mmbtu"]) > 10)
+        | (abs(plant_total_diff["net_generation_mwh"]) > 10)
+    ]
+    if len(mismatched_allocation) > 0:
```

Comment:
Are there currently any plants where this is an issue?

Reply:
There are, but none of the magnitudes looked huge, and I don't think we have the bandwidth to track down the issues with each individual plant now. 

For example, here is the output for 2021:

```
WARNING: Allocated EIA-923 doesn't match input data for plants:
              net_generation_mwh  fuel_consumed_mmbtu  fuel_consumed_for_electricity_mmbtu
plant_id_eia                                                                              
1206                   60.568183          1241.475753                          1241.475753
1316                  -48.000000             0.000000                             0.000000
52148                   1.601000            18.000000                             7.000000
54224                  14.960000           150.000000                            66.000000
54967                   1.960000            24.000000                            24.000000
58256                -241.000000             0.000000                             0.000000
59260                 836.856000         13729.000000                         13729.000000
59825                9539.000000         84354.000000                         84354.000000
```
Looking at the two plants at the bottom, the missing data from 59260 is because the plant retires mid year (according to EIA-860), but there is still reported fuel consumption after that. Plant 59825 is a solar plant that has some wind generation reported in the gf table. These are things that we should look into, but they probably requiring tracking a lot of specific edge cases, and none of the magnitudes are that great. 


However, maybe we should consider exporting a "warnings.txt" file that saves all of the outputs of the warning messages for each run so that we can easily track the data quality issues that were flagged but ignored for each run.